prow/cmd/checkconfig/main.go

```diff
@@ -1283,3 +1276,26 @@ func prefixWithAndIfNeeded(s string, needsAnd bool) string {
 	}
 	return s
 }
+
+func getSupplementalConfigScope(path string, filesystem fs.FS, supplementalProwConfigsFileNameSuffix string) (global bool, orgs sets.String, repos sets.String, err error) {
+
+	var cfg interface {
+		HasConfigFor() (bool, sets.String, sets.String)
+	}
+	if strings.HasSuffix(path, supplementalProwConfigsFileNameSuffix) {
+		cfg = &config.Config{}
+	} else {
```

Comment:
This is a weak inference and requires that the caller continue to run the validation before calling this - might be nice to pass in an enum or the type to marshal into itself and not expose this function to the suffixes?